### PR TITLE
update minimum version to 2020.3.42

### PR DESCRIPTION
**Pull request by davidkline-ms**
This change sets the project version to 2020.3.42 which contains the fix for IL2CPP builds not compiling using Visual Studio 2022 17.4.

### CODE REVIEW

**Comment by Zee2** [comment]
From what I can tell, this isn't actually updating the minimum Unity version(s) we require in our individual package manifests?

**Comment by davidkline-ms** [comment]
> From what I can tell, this isn't actually updating the minimum Unity version(s) we require in our individual package manifests?

This is true. We do not currently enforce a minimum revision in the manifests (this is not something we have previously done).

**Comment by MaxWang-MS** [comment]
Shall we also update the readme page?

**Comment by MaxWang-MS** [comment]
the readme at the repo root.

**Comment by davidkline-ms** [comment]
> Shall we also update the readme page?

added to the readme.md in the mrtk3 branch root

**Comment by davidkline-ms** [comment]
> From what I can tell, this isn't actually updating the minimum Unity version(s) we require in our individual package manifests?

added. please note that MRFT will need an update to actually use this data.

**Comment by MaxWang-MS** [comment]
I am going to say some unpopular thing here... I do not think we should be enforcing the min Unity patch version for our packages. There are all kinds of reasons why people cannot upgrade beyond a specific version of Unity (due to Unity/external bug, for example). I am not sure whether the newly added `unityRelease` field does anything (e.g., prevents user from importing our tarball into a older patch version of Unity) without an MRFT update. But nevertheless, I don't think we should be dictating to that extent, especially given this Unity bug is workaroundable/does not impact everyone. @davidkline-ms @Zee2

**Comment by Victor-x1** [comment]
The current MRTK3 DevTemplate successfully loads in 2021.3.14f without errors.

**Comment by davidkline-ms** [comment]
> I am going to say some unpopular thing here... I do not think we should be enforcing the min Unity patch version for our packages. There are all kinds of reasons why people cannot upgrade beyond a specific version of Unity (due to Unity/external bug, for example). I am not sure whether the newly added `unityRelease` field does anything (e.g., prevents user from importing our tarball into a older patch version of Unity) without an MRFT update. But nevertheless, I don't think we should be dictating to that extent, especially given this Unity bug is workaroundable/does not impact everyone. @davidkline-ms @Zee2

I am ok removing this entry in the json files. I agree there are some mitigating scenarios where it would not be ideal to have the release version specified and will 100% go with the consensus.

**Comment by keveleigh** [comment]
> I am not sure whether the newly added unityRelease field does anything (e.g., prevents user from importing our tarball into a older patch version of Unity) without an MRFT update

Yeah, I _think_ it's mainly used by the UPM UI for displaying relevant versions from a registry and _not_ for preventing import. I think the `unity` field is used the same way.

**Comment by keveleigh** [comment]
> I do not think we should be enforcing the min Unity patch version for our packages.

I think I agree with this. Since this is less a "we use an API introduced in a Unity patch release and our package won't build before this Unity version" (which happens, unfortunately 😅) and more just "Unity and Visual Studio have an incompatibility with IL2CPP prior to this version", I'm fine leaving it out and treating this as something to heavily document.

**Comment by davidkline-ms** [comment]
> I'm fine leaving it out and treating this as something to heavily document.

That works for me. I will gladly remove the release tags from the manifests.

File: README.md [hunk 1]
```diff
@@ -82,4 +82,4 @@ The roadmap from public preview to GA is detailed in the following table.
 | --- | --- |
 | Public Preview | June 8, 2022 |
 | Preview updates | Approximately every 2-4 weeks until GA |
-| General Availability | Fall / Winter 2022 |
+| General Availability | First quarter of  2023 |
```
Comment (keveleigh) [inline README.md:85]:
```suggestion
| General Availability | First quarter of 2023 |
```

**Review by keveleigh — APPROVED** [review]

**Review by MaxWang-MS — APPROVED** [review]

**Review by MaxWang-MS — APPROVED** [review]

**Review by keveleigh — APPROVED** [review]

**Pull request merged**